### PR TITLE
Add support for inline image

### PR DESCRIPTION
**Pull request by tomekzaw**
### Details

This PR adds support for inline images inserted with the following syntax: `alt text`.

### Related Issues

https://github.com/Expensify/App/issues/39275

### CODE REVIEW

File: parser/__tests__/index.test.js [hunk 1]
```diff
@@ -378,3 +378,90 @@ describe('trailing whitespace', () => {
...
+      {type: 'syntax', start: 29, length: 2},
+      {type: 'link', start: 31, length: 29},
+      {type: 'syntax', start: 60, length: 1},
+    ]);
+  });
+
+  test('inline code in alt', () => {
+    // TODO: fix this test
+    expect('![`code`](https://example.com/image.png)').toBeParsedAs([]);
+  });
```
Comment (tomekzaw) [inline parser/__tests__/index.test.js:461]: Currently, this test fails

File: parser/__tests__/index.test.js [hunk 2]
```diff
@@ -378,3 +378,90 @@ describe('trailing whitespace', () => {
...
+
+  test('blockquote in alt', () => {
+    // TODO: confirm the expected value
+    expect('![```blockquote```](https://example.com/image.png)').toBeParsedAs([]);
+  });
+});
```
Comment (tomekzaw) [inline parser/__tests__/index.test.js:467]: Is this really the expected result?

File: parser/__tests__/index.test.js [hunk 3]
```diff
@@ -378,3 +378,128 @@ describe('trailing whitespace', () => {
...
+  test('without alt text', () => {
+    expect('![](https://example.com/image.png)').toBeParsedAs([
+      {type: 'syntax', start: 0, length: 1},
+      {type: 'syntax', start: 1, length: 1},
+      {type: 'syntax', start: 2, length: 1},
+      {type: 'syntax', start: 3, length: 1},
+      {type: 'link', start: 4, length: 29},
+      {type: 'syntax', start: 33, length: 1},
+    ]);
+  });
```
Comment (tomekzaw) [inline parser/__tests__/index.test.js:403]: Let's add a newline here
```suggestion
  });

```

File: parser/__tests__/index.test.js [hunk 4]
```diff
@@ -378,3 +378,128 @@ describe('trailing whitespace', () => {
...
+      {type: 'syntax', start: 1, length: 1},
+      {type: 'syntax', start: 8, length: 1},
+      {type: 'syntax', start: 9, length: 1},
+      {type: 'link', start: 10, length: 29},
+      {type: 'syntax', start: 39, length: 1},
+    ]);
+  });
+
+  test('blockquote in alt', () => {
+    // TODO: this test is failing !!!
```
Comment (tomekzaw) [inline parser/__tests__/index.test.js:486]: Should we mark it as "xfail" or just fix it?

File: parser/index.ts [hunk 5]
```diff
@@ -158,6 +162,20 @@ function parseTreeToTextAndRanges(tree: StackItem): [string, Range[]] {
           addChildrenWithStyle(matchString, 'link');
           appendSyntax(')');
         }
+      } else if (node.tag.startsWith('<img src="')) {
+        const src = node.tag.match(/src="([^"]*)"/)![1]!; // always present
+        const alt = node.tag.match(/alt="([^"]*)"/);
+        const isLabeledLink = node.tag.match(/data-link-variant="([^"]*)"/)![1] === 'labeled';
```
Comment (tomekzaw) [inline parser/index.ts:168]:
```suggestion
        const hasAlt = node.tag.match(/data-link-variant="([^"]*)"/)![1] === 'labeled';
```

File: parser/index.ts [hunk 6]
```diff
@@ -158,6 +162,20 @@ function parseTreeToTextAndRanges(tree: StackItem): [string, Range[]] {
...
+
+        appendSyntax('!');
+        if (isLabeledLink) {
```
Comment (tomekzaw) [inline parser/index.ts:171]:
```suggestion
        if (hasAlt) {
```

File: parser/__tests__/index.test.js (hunk 4 above)
Comment (robertKozik) [reply parser/__tests__/index.test.js]: Actually this comment should be removed, test is already fixed

**Review by robertKozik — DISMISSED** [review]
👀 ✅ 👀

**Comment by tomekzaw** [comment]
It seems like there's some problem with the following input:

Screenshot 2024-04-09 at 00 19 41

File: parser/__tests__/index.test.js [hunk 7]
```diff
@@ -502,4 +502,17 @@ describe('inline image', () => {
       {type: 'syntax', start: 31, length: 1},
     ]);
   });
+
+  test('image with empty alt text and not completed link', () => {
+    expect('# ![](onet.pl)').toBeParsedAs([
```
Comment (tomekzaw) [inline parser/__tests__/index.test.js:507]:
```suggestion
    expect('# ![](example.com)').toBeParsedAs([
```

**Review by robertKozik — APPROVED** [review]

**Pull request merged**